File: day3/factorial.py
def factorialIterative(n):
    if(n == 0):
        return 1
    
    fact = 1
    for i in range (1, n + 1):
        fact *= i
        
    return fact

def factorialRecursive(n):
    if (n == 0):
        return 1
    
    if(n == 1):
        return 1
    else:
        return n * factorialRecursive(n - 1)
    
n = int(input("Enter a no.: "))
res1 = factorialIterative(n)
res2 = factorialRecursive(n)

print(f"Factorial of {n} by Iterative method = {res1}")
print(f"Factorial of {n} by Recursive method = {res2}")

File: day5/factorial.r
factorial_iterative <- function(n) {
  fact <- 1
  if (n == 0) {
    return(fact)
  }

  for (i in 1:n) {
    fact <- fact * i
  }
  fact
}

factorial_recursive <- function(n) {
  if (n == 0) {
    1
  } else if (n == 1) {
    1
  } else {
    n * factorial_recursive(n - 1)
  }
}

n <- as.integer(readline("Enter a no.: "))
cat("Factorial by Iterative Method =", factorial_iterative(n), "\n")
cat("Factorial by Recursive Method =", factorial_recursive(n), "\n")